editor/src/components/navigator/navigator-item/navigator-item-dnd-container.tsx

```diff
@@ -547,24 +535,32 @@ export const NavigatorItemContainer = React.memo((props: NavigatorItemDragAndDro
   )
 
   const isFirstSibling = React.useMemo(() => {
-    const siblings = MetadataUtils.getSiblingsOrdered(metadata, elementPathTree, props.elementPath)
+    const siblings = MetadataUtils.getSiblingsOrdered(
+      editorStateRef.current.jsxMetadata,
+      editorStateRef.current.elementPathTree,
+      props.elementPath,
+    )
     const firstSibling = siblings.at(0)
     if (firstSibling == null) {
       return false
     }
 
     return EP.pathsEqual(firstSibling.elementPath, props.elementPath)
-  }, [metadata, elementPathTree, props.elementPath])
+  }, [editorStateRef, props.elementPath])
 
   const isLastSibling = React.useMemo(() => {
-    const siblings = MetadataUtils.getSiblingsOrdered(metadata, elementPathTree, props.elementPath)
+    const siblings = MetadataUtils.getSiblingsOrdered(
```

Comment:
Does this work in the case where reparenting something in the navigator should cause this to change? Like if I dragged something from elsewhere to the end of a few siblings? Although as I write this I suspect it does work because any change to the hierarchy would cause the entire navigator to re-render.

I think this might be fine in the whole, but I'm a little wary of it because the dependencies aren't right locally and some change to how we render the navigator could cause this to render incorrectly. At the very least there should be a warning comment for our future selves.

Reply:
good shout! it seems to be working fine for me but I'll add a comment

Reply:
https://github.com/concrete-utopia/utopia/pull/4055/commits/db0af6abb8d711b7b17f7a4ff44b9a38ee30d97a and https://github.com/concrete-utopia/utopia/pull/4055/commits/15ef8b6c8b7584300a93e924172f3f25d7dce611